opensearch-operator/pkg/builders/cluster.go

```diff
@@ -136,6 +144,23 @@ func NewSTSForNodePool(cr *opsterv1.OpenSearchCluster, node opsterv1.NodePool, v
 		ProbeHandler:        corev1.ProbeHandler{TCPSocket: &corev1.TCPSocketAction{Port: intstr.IntOrString{IntVal: cr.Spec.General.HttpPort}}},
 	}
 
+	// Because the http endpoint requires auth we need to do it as a curl script
+	readinessProbe := corev1.Probe{
+		InitialDelaySeconds: 30,
+		PeriodSeconds:       30,
+		ProbeHandler: corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{
+				Command: []string{
+					"/bin/bash",
+					"-c",
+					"curl -k -u ${OS_USER}:${OS_PASSWORD} --silent --fail https://localhost:9200",
```

Comment:
I would prefer `OPENSEARCH_USER` and `OPENSEARCH_PASSWORD` just to avoid the potential confusion over the double meaning of `OS`.